internal/source/git.go

```diff
@@ -126,8 +107,40 @@ func (r *Git) Unpack(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (*Resul
 	return &Result{Bundle: bundleFS, ResolvedSource: resolvedSource, State: StateUnpacked}, nil
 }
 
+func (r *Git) createCloneOptions(bundle *rukpakv1alpha1.Bundle, progress bytes.Buffer) (*git.CloneOptions, error) {
+	gitsource := bundle.Spec.Source.Git
+	if gitsource.Repository == "" {
+		// This should never happen because the validation webhook rejects git bundles without repository
+		return nil, errors.New("missing git source information: repository must be provided")
+	}
+
+	cloneOpts := git.CloneOptions{
+		URL:             gitsource.Repository,
+		Progress:        &progress,
+		Tags:            git.NoTags,
+		InsecureSkipTLS: bundle.Spec.Source.Git.Auth.InsecureSkipVerify,
+	}
+
+	if gitsource.Ref.Branch != "" && gitsource.Ref.Tag != "" {
+		return nil, errors.New("git branch and tag cannot be set simultaneously")
+	}
+	if gitsource.Ref.Branch != "" {
+		cloneOpts.ReferenceName = plumbing.ReferenceName(fmt.Sprintf("refs/heads/%s", gitsource.Ref.Branch))
+		cloneOpts.SingleBranch = true
+		cloneOpts.Depth = 1
+	} else if gitsource.Ref.Tag != "" {
```

Comment:
nit: we can improve readability here if we remove the if/else conditional structure, and exit early in the if statement by returning the cloneOpts variable there.

Reply:
makes sense, but I did it this way so we could validate against setting Branch and Tag. If we return after Branch, then Branch+Tag simultaneously is legal, but Branch wins.